### PR TITLE
SUS-3446 Fix QuickTools rollback feature

### PR DESCRIPTION
**Pull request by mszabo-wikia**
Same as https://github.com/Wikia/app/pull/14312

https://wikia-inc.atlassian.net/browse/SUS-3446

Same in English

### CODE REVIEW

File: extensions/wikia/SyntaxHighlight/tests/SyntaxHighlightTest.php [hunk 1]
```diff
@@ -17,6 +17,8 @@ protected function setUp() {
 		$this->setupFile = __DIR__ . '/../SyntaxHighlight.setup.php';
 		parent::setUp();
 
+		Hooks::register( 'ParserFirstCallInit', 'SyntaxHighlightHooks::onParserFirstCallInit' );
+
```
Comment (macbre) [inline extensions/wikia/SyntaxHighlight/tests/SyntaxHighlightTest.php:21]: ?

File: includes/db/Database.php [hunk 2]
```diff
@@ -3606,6 +3606,18 @@ protected function indexNameCallback( $matches ) {
...
+	 *
+	 * @param string $lockName name of lock to poll
+	 * @param string $method name of method calling us
+	 * @return Boolean
+	 * @since 1.20
+	 */
+	public function lockIsFree( $lockName, $method ) {
+		return true;
+	}
+
```
Comment (macbre) [inline includes/db/Database.php:3620]: ?

File: extensions/wikia/SyntaxHighlight/tests/SyntaxHighlightTest.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/wikia/SyntaxHighlight/tests/SyntaxHighlightTest.php:21]: We've unset the hooks in other test, so it's better to register it here

File: includes/db/Database.php (hunk 2 above)
Comment (mszabo-wikia) [reply includes/db/Database.php:3620]: This method was backported from 1.20 but only to `DatabaseMysqlBase` class. It is thus missing from the parent class and causes an error if something tries to call it in SQLite integration tests.

**Review by macbre — APPROVED** [review]
Noticed two "strange" changes

**Pull request merged**